src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1496,13 +1497,20 @@ private List<EC2AbstractSlave> provisionSpot(Image image, int number, EnumSet<Pr
 
             launchSpecification.setUserData(userDataString);
 
+            KeyPair keypair = null;
 
             if (!isUsingDynamicSshKeys()) {
                 // there is a global/static ssh keypair configured, so use that
-                KeyPair kp = getParent().resolveKeyPair();
-                if (kp != null) {
-                    launchSpecification.setKeyName(kp.getKeyName());
+                LOGGER.fine(() -> "static ssh credential is configured, resolving keyname to use in spot launch request");
+                keypair = getParent().resolveKeyPair();
+                if (keypair != null) {
+                    launchSpecification.setKeyName(keypair.getKeyName());
                 }
+            } else {
+                String keyName = "dev-1-" + UUID.randomUUID();
```

Comment:
Is this supposed to be temporary code?

Reply:
yes, i did not mean to commit that change :face_exhaling: